simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/handlers/ocpp/CancelReservationRequestHandler.java

```diff
@@ -22,37 +23,27 @@
     @Override
     public void handle(String callId, CancelReservationRequest request) {
         Optional<Reservation> reservationOpt = stationStore.tryFindReservationById(request.getReservationId());
+        CancelReservationResponse cancelReservationResponse = reservationOpt.map(reservation -> new CancelReservationResponse().withStatus(CancelReservationResponse.Status.ACCEPTED))
+                .orElseGet(() -> new CancelReservationResponse().withStatus(CancelReservationResponse.Status.REJECTED));
 
-        if (!reservationOpt.isPresent()) {
-            stationMessageSender.sendCallResult(callId, new CancelReservationResponse().withStatus(CancelReservationResponse.Status.REJECTED));
-        } else {
-            Reservation reservation = reservationOpt.get();
-            stationMessageSender.sendCallResult(callId, new CancelReservationResponse().withStatus(CancelReservationResponse.Status.ACCEPTED));
-
-            if (isConnectorReserved(reservation)) {
+        reservationOpt.ifPresent(reservation -> {
+            if (reservation.getEvse().getConnectorId() != null && isConnectorReserved(reservation)) {
                 makeConnectorAvailable(reservation);
                 stationMessageSender.sendStatusNotification(reservation.getEvse().getId().intValue(), reservation.getEvse().getConnectorId().intValue(), StatusNotificationRequest.ConnectorStatus.AVAILABLE);
             }
-
             stationStore.removeReservation(reservation);
-        }
+        });
+        stationMessageSender.sendCallResult(callId, cancelReservationResponse);
     }
 
     private boolean isConnectorReserved(Reservation reservation) {
-        if (reservation.getEvse().getConnectorId() != null)  {
-            Optional<Connector> connector = stationStore.tryFindConnector(reservation.getEvse().getId(), reservation.getEvse().getConnectorId());
-            if (connector.isPresent()) {
-                return StatusNotificationRequest.ConnectorStatus.RESERVED.equals(connector.get().getConnectorStatus());
-            }
-        }
-        return false;
+        return stationStore.tryFindConnector(reservation.getEvse().getId(), reservation.getEvse().getConnectorId())
+                    .map(connector -> StatusNotificationRequest.ConnectorStatus.RESERVED.equals(connector.getConnectorStatus()))
```

Comment:
nit piking: this map function actually consists of 2 action and could be redesigned as
```
.map(Connector::getConnectorStatus())
.map(status -> StatusNotificationRequest.ConnectorStatus.RESERVED.equals(status))
```

for this particular case it doesn't really matter, but for a larger flow it could be useful

Reply:
Updated